### ISSUE TITLE
Support for struct

### ISSUE THREAD

**Issue by Socolin** [diegofrata/Generator.Equals#28]
Hello,

Would it be possible to also generate equality members for `struct` ?

**Comment by diegofrata** [comment]
Yes, although I can't think of a scenario where I would ever use it. Out of curiosity, how would you use it?

**Comment by Socolin** [comment]
I have some enums like this one and I would like to generate the equality members automatically

```cs
[Equatable]
public readonly struct Color
{
	[DefaultEquality]
	public int R { get; }
	[DefaultEquality]
	public int G { get; }
	[DefaultEquality]
	public int B { get; }
	[DefaultEquality]
	public int A { get; }
}
```

**Comment by diegofrata** [comment]
That should already be equatable by default -- as long as every field in a struct is a value type, you should get the same semantics as records. The only caveat is that, apparently, it uses reflection which is slower. I think it's easy enough to add support to structs, I will have a look. 

By the way, you only need to specify `DefaultEquality` if using explicit mode. This would be shorter:

```cs
[Equatable]
public readonly struct Color
{
	public int R { get; }
	public int G { get; }
	public int B { get; }
	public int A { get; }
}
```

**Comment by diegofrata** [comment]
This is now available in version 2.6.0
